tests/array_fields.rs

```diff
@@ -0,0 +1,88 @@
+use derive_mmio::OutOfBoundsError;
+
+#[derive(derive_mmio::Mmio)]
+#[repr(C)]
+struct Uart {
+    control: u32,
+    some_array_0: [u32; 4],
+    other_field: u32,
+    some_array_1: [u32; 2],
+}
+
+fn main() {
+    let mut uart = Uart {
+        control: 0xC,
+        other_field: 0,
+        some_array_0: [0x1, 0x2, 0x3, 0x4],
+        some_array_1: [0x12, 0x44],
+    };
+
+    // Safety: We're pointing at a real object
+    let mut mmio_uart = unsafe { Uart::new_mmio(core::ptr::addr_of_mut!(uart)) };
+    assert_eq!(mmio_uart.read_some_array_0(0).unwrap(), 0x1);
+    assert_eq!(mmio_uart.read_some_array_0(1).unwrap(), 0x2);
+    assert_eq!(mmio_uart.read_some_array_0(2).unwrap(), 0x3);
+    assert_eq!(mmio_uart.read_some_array_0(3).unwrap(), 0x4);
+    unsafe {
+        assert_eq!(mmio_uart.read_some_array_0_unchecked(0), 0x1);
+        assert_eq!(mmio_uart.read_some_array_0_unchecked(1), 0x2);
+        assert_eq!(mmio_uart.read_some_array_0_unchecked(2), 0x3);
+        assert_eq!(mmio_uart.read_some_array_0_unchecked(3), 0x4);
+    }
+
+    assert_eq!(mmio_uart.read_some_array_1(0).unwrap(), 0x12);
+    assert_eq!(mmio_uart.read_some_array_1(1).unwrap(), 0x44);
+    unsafe {
+        assert_eq!(mmio_uart.read_some_array_1_unchecked(0), 0x12);
+        assert_eq!(mmio_uart.read_some_array_1_unchecked(1), 0x44);
+    }
+
+    for (idx, val) in (0..3).rev().enumerate() {
```

Comment:
This only tests three elements of the array (index 0, 1, and 2). That might not be intentional.